From d9e95decc35de4658bf3a2918b10dc4220926352 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 19 Sep 2014 12:29:15 -0400
Subject: [acpi] hotplug/pci: Simplify acpi_install_hotplug_notify_handler()

Message-id: <1411129776-21656-41-git-send-email-prarit@redhat.com>
Patchwork-id: 96357
O-Subject: [RHEL7.1 PATCH BZ 1141846 40/61] ACPI / hotplug / PCI: Simplify acpi_install_hotplug_notify_handler()
Bugzilla: 1141846
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Myron Stowe <mstowe@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1141846

commit 5e6f236c263117cef5f0d68e3fec241ba2adc4fc
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Thu Feb 6 13:57:58 2014 +0100

    ACPI / hotplug / PCI: Simplify acpi_install_hotplug_notify_handler()

    Since acpi_hotplug_notify_cb() does not use its data argument any
    more, the second argument of acpi_install_hotplug_notify_handler()
    can be dropped, so do that and update its callers accordingly.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
Cc: Stefan Assmann <sassmann@redhat.com>
Cc: Myron Stowe <mstowe@redhat.com>
Cc: Andy Gospodarek <agospoda@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/scan.c b/drivers/acpi/scan.c
index ed8162f..3b16d58 100644
--- a/drivers/acpi/scan.c
+++ b/drivers/acpi/scan.c
@@ -519,10 +519,10 @@ static void acpi_hotplug_notify_cb(acpi_handle handle, u32 type, void *data)
  acpi_evaluate_hotplug_ost(handle, type, ost_code, NULL);
 }
 
-void acpi_install_hotplug_notify_handler(acpi_handle handle, void *data)
+void acpi_install_hotplug_notify_handler(acpi_handle handle)
 {
  acpi_install_notify_handler(handle, ACPI_SYSTEM_NOTIFY,
-        acpi_hotplug_notify_cb, data);
+        acpi_hotplug_notify_cb, NULL);
 }
 
 void acpi_remove_hotplug_notify_handler(acpi_handle handle)
@@ -1983,7 +1983,7 @@ static void acpi_scan_init_hotplug(struct acpi_device *adev)
   if (!handler)
    continue;
 
-  acpi_install_hotplug_notify_handler(adev->handle, handler);
+  acpi_install_hotplug_notify_handler(adev->handle);
   adev->flags.hotplug_notify = true;
   break;
  }
diff --git a/drivers/pci/hotplug/acpiphp_glue.c b/drivers/pci/hotplug/acpiphp_glue.c
index 55ac45f..205af1a 100644
--- a/drivers/pci/hotplug/acpiphp_glue.c
+++ b/drivers/pci/hotplug/acpiphp_glue.c
@@ -376,7 +376,7 @@ static acpi_status register_slot(acpi_handle handle, u32 lvl, void *data,
 
  /* install notify handler */
  if (!(newfunc->flags & FUNC_HAS_DCK))
-  acpi_install_hotplug_notify_handler(handle, NULL);
+  acpi_install_hotplug_notify_handler(handle);
 
  return AE_OK;
 }
diff --git a/include/acpi/acpi_bus.h b/include/acpi/acpi_bus.h
index 7781e20..3170914 100644
--- a/include/acpi/acpi_bus.h
+++ b/include/acpi/acpi_bus.h
@@ -429,7 +429,7 @@ static inline bool acpi_device_enumerated(struct acpi_device *adev)
 typedef void (*acpi_hp_callback)(void *data, u32 src);
 
 acpi_status acpi_hotplug_execute(acpi_hp_callback func, void *data, u32 src);
-void acpi_install_hotplug_notify_handler(acpi_handle handle, void *data);
+void acpi_install_hotplug_notify_handler(acpi_handle handle);
 void acpi_remove_hotplug_notify_handler(acpi_handle handle);
 
 /**
-- 
1.7.1